### PR TITLE
Moving icon toggle to the v2 component model.

### PR DESCRIPTION
**Pull request by sgomes**
Very similar to checkbox and friends.

Once again, minimal changes to CSS/DOM, as that's better handled in separate PRs if need be.

@surma and/or @Garbee: PTAL!

### CODE REVIEW

File: src/icon-toggle/icon-toggle.js [hunk 1]
```diff
@@ -15,242 +15,215 @@
...
+   * Attach all listeners to the DOM.
    *
-   * @private
+   * @export
    */
-  MaterialIconToggle.prototype.onBlur_ = function() {
-    this.element_.classList.remove(this.CssClasses_.IS_FOCUSED);
-  };
+  addEventListeners() {
+    this.input_.addEventListener('change', this.changeListener_);
```
Comment (surma) [inline src/icon-toggle/icon-toggle.js]: Should these be bound to `this`?

Comment (surma) [inline src/icon-toggle/icon-toggle.js]: nvm me

**Comment by surma** [comment]
LGTM :)

**Comment by addyosmani** [comment]
LGTM2

**Comment by sgomes** [comment]
Thanks, guys! 👍

**Comment by Garbee** [comment]
LGTM as well. One issue but it is trivial and we will eventually cover it anyways once the new simple-mode thing is worked on. (Existence of `mdl-js-icon-toggle` in the registered CSS Classes property.)

**Comment by sgomes** [comment]
Yeah, I think we should do a pass and remove all the `js` classes at some point, or just wait for the simple mode PR and handle it all there. Thanks, @Garbee!

**Pull request merged**